app/lib/actions/katello/repository/metadata_generate.rb

```diff
@@ -10,15 +10,15 @@ def plan(repository, options = {})
           repository_creation = options.fetch(:repository_creation, false)
           source_repository = options.fetch(:source_repository, nil)
           source_repository ||= repository.target_repository if repository.link?
-          smart_proxy_id = options.fetch(:capsule_id, SmartProxy.pulp_master.id)
+          smart_proxy = options.fetch(:smart_proxy, SmartProxy.pulp_master)
           if repository_creation
             matching_content = false
           else
             matching_content = options.fetch(:matching_content, false)
           end
 
           plan_pulp_action([Pulp::Repository::DistributorPublish, Pulp3::Orchestration::Repository::GenerateMetadata],
-                        repository, SmartProxy.find(smart_proxy_id),
+                        repository, smart_proxy,
                         :force => force,
                         :source_repository => source_repository,
                         :matching_content => matching_content,
```

Comment:
@jlsherrill : I remember changing this earlier. Is this is a refactor or was this causing the issue?

Reply:
SmartProxy.find() respects location and permission settings, so we can't ever use it as is in these types of actions (but instead need to use .unscoped).  In this case the smart proxy id was being passed in, and had already been looked up, so i figured we could just use it as is